server/src/main/java/org/elasticsearch/action/search/ExpandSearchPhase.java

```diff
@@ -91,7 +91,8 @@ public void run() throws IOException {
                     SearchSourceBuilder sourceBuilder = buildExpandSearchSourceBuilder(innerHitBuilder, innerCollapseBuilder)
                         .query(groupQuery)
                         .postFilter(searchRequest.source().postFilter());
-                    SearchRequest groupRequest = buildExpandSearchRequest(searchRequest, sourceBuilder);
+                    SearchRequest groupRequest = new SearchRequest(searchRequest);
+                    groupRequest.source(sourceBuilder);
```

Comment:
Looking at https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/action/search/SearchRequest.java#L112 I think this extra line can be removed, or am I missing sth?

Reply:
if we remove it the copy constructor would not be a correct copy constructor, meaning it would never copy the source? I think it's fine that it copies it and we override it when needed? WDYT?

Reply:
Oh, I wasn't suggesting removing it in the copy constructor. I was missing the part that `sourceBuilder` here seems not be part of `searchRequest` yet when calling the copy constructor. All good now, thanks for the clarification.